pytools/py_codegen.py

```diff
@@ -35,12 +35,18 @@
 
 
 class PythonCodeGenerator(CodeGeneratorBase):
-    def get_module(self, name=None):
+    def get_module(self, name=None, write_file=True):
```

Comment:
Could maybe set this to False by default.

Reply:
That won't work for the line_profiler unfortunately, it requires the code in a file, even if it is in the linecache: https://github.com/pyutils/line_profiler/blob/1630e7c9a295ace2feb1d2b188e68f4d2833fb20/line_profiler/line_profiler.py#L194-L210

Reply:
[51ada01](https://github.com/inducer/pytools/pull/295/commits/51ada018e4b2bb95203a2856505e520040ca1e28) implements a workaround that uses linecache directly (like you suggested), but also maintains compatibility with line_profiler.

Reply:
( I added a warning, it is somewhat noisy though)